### PR TITLE
DSDEGP-2059 Set ServerName to real hostname, and ServerAlias to DNS.

### PR DESCRIPTION
**Pull request by danxmoran**
### Description

This will allow us to refer to deployed Clio servers using both the DNS and their specific hostname. Currently, we can only use the DNS (otherwise the SSL handshake fails in the client).

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: Ensure that you have added or updated tests
- [ ] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

File: jenkins/deploy-clio-server.sh [hunk 1]
```diff
@@ -115,6 +115,22 @@ compute_docker_tag() {
...
 
+# Build a "conventional" fully-qualified domain name from a hostname and environment.
+build_fqdn() {
+  echo ${1}.gotc-${ENV}.broadinstitute.org
+}
+
+# Get the "real" hostname for the Clio host at the given address (which might be a DNS).
+# Assumes that Clio hosts will be named according to the pattern:
+#  gotc-clio-{ENV}(\d{3})
+# i.e. gotc-clio-dev101, gotc-clio-prod-203
```
Comment (gbggrant) [inline jenkins/deploy-clio-server.sh]: in this example is there really a '-' in gotc-clio-prod-203?  Then I think the sed below won't work for that?

File: jenkins/deploy-clio-server.sh [hunk 2]
```diff
@@ -115,6 +115,22 @@ compute_docker_tag() {
...
+get_real_clio_name() {
+  local -r gce_name=$(ssh ${SSH_OPTS[@]} ${SSH_USER}@$1 'uname -n')
+  local -r instance_number=$(echo ${gce_name} | sed "s/gotc-clio-${ENV}\([0-9]*\)/\1/")
```
Comment (gbggrant) [inline jenkins/deploy-clio-server.sh]: Can't do \d{3} here?  (rather than [0-9]*)

**Review by gbggrant — APPROVED** [review]
👍

**Review by charlesdyan — APPROVED** [review]

**Review by pshapiro4broad — APPROVED** [review]

File: jenkins/deploy-clio-server.sh (hunk 2 above)
Comment (danxmoran) [reply jenkins/deploy-clio-server.sh]: Nope, didn't work.

Comment (danxmoran) [reply jenkins/deploy-clio-server.sh]: {3} works with extended RE, though!

**Pull request merged**